### PR TITLE
Fix default install tag for shared lib symlinks

### PR DESCRIPTION
**Pull request by xclaesse**
Versioned shared libraries should have .so file in devel, .so.1 and
.so.1.2.3 in runtime.

### CODE REVIEW

File: mesonbuild/minstall.py [hunk 1]
```diff
@@ -463,8 +463,8 @@ def do_symlink(self, target: str, link: str, full_dst_dir: str) -> bool:
         abs_target = target
         if not os.path.isabs(target):
             abs_target = os.path.join(full_dst_dir, target)
-        if not os.path.exists(abs_target):
-            raise RuntimeError(f'Tried to install symlink to missing file {abs_target}')
+        #if not os.path.exists(abs_target):
+        #    raise RuntimeError(f'Tried to install symlink to missing file {abs_target}')
```
Comment (xclaesse) [inline mesonbuild/minstall.py:467]: This is a TODO: when installing only `devel` the target does not exist, not sure we should always allow installing symlink to a target that does not exist??

File: mesonbuild/mintro.py [hunk 2]
```diff
@@ -107,9 +107,6 @@ def list_installed(installdata: backends.InstallData) -> T.Dict[str, str]:
         for t in installdata.targets:
             res[os.path.join(installdata.build_dir, t.fname)] = \
                 os.path.join(installdata.prefix, t.outdir, os.path.basename(t.fname))
-            for alias in t.aliases.keys():
-                res[os.path.join(installdata.build_dir, alias)] = \
-                    os.path.join(installdata.prefix, t.outdir, os.path.basename(alias))
```
Comment (xclaesse) [inline mesonbuild/mintro.py:112]: TODO: aliases are now parts of `installdata.symlinks` but those are not part of introspection data.

File: mesonbuild/minstall.py (hunk 1 above)
Comment (eli-schwartz) [inline mesonbuild/minstall.py:467]: Huh, why does this need to be commented out? Don't we install destinations before symlinks already?

File: mesonbuild/mintro.py [hunk 3]
```diff
@@ -217,7 +214,6 @@ def list_targets(builddata: build.Build, installdata: backends.InstallData, back
     install_lookuptable = {}
     for i in installdata.targets:
         out = [os.path.join(installdata.prefix, i.outdir, os.path.basename(i.fname))]
-        out += [os.path.join(installdata.prefix, i.outdir, os.path.basename(x)) for x in i.aliases]
```
Comment (xclaesse) [inline mesonbuild/mintro.py:220]: TODO: should symlinks be part of targets?

File: mesonbuild/minstall.py (hunk 1 above)
Comment (xclaesse) [reply mesonbuild/minstall.py:467]: Opened a TODO review comment myself about this, see above.

Comment (eli-schwartz) [reply mesonbuild/minstall.py:467]: Ah, good point!

Comment (xclaesse) [reply mesonbuild/minstall.py]: Allowed it only for library symlinks for now. Better be conservative.

File: mesonbuild/mintro.py (hunk 2 above)
Comment (xclaesse) [reply mesonbuild/mintro.py:112]: Side effect of this PR, installdata.symlinks are now part of introspection data.

File: mesonbuild/mintro.py (hunk 3 above)
Comment (xclaesse) [reply mesonbuild/mintro.py:220]: yes, now they are.

**Comment by xclaesse** [comment]
@eli-schwartz ok, finally got the CI pass here, ready for review \o/

**Pull request merged**